build.gradle

```diff
@@ -146,11 +147,131 @@ project(':api') {
+                    'Built-By': 'MinIO, inc',
+                    'Built-JDK': System.getProperty('java.version'),
+                    'Source-Compatibility': sourceCompatibility,
+                    'Target-Compatibility': targetCompatibility)
+        }
+    }
+
+    java {
+        withJavadocJar()
+        withSourcesJar()
+    }
+
+    javadoc.options {
+        encoding = 'UTF-8'
+        if (JavaVersion.current() == JavaVersion.VERSION_1_8) {
+            links 'https://docs.oracle.com/javase/8/docs/api/'
+        } else {
+            links 'https://docs.oracle.com/en/java/javase/11/docs/api/'
+        }
+    }
+
+    // Disable JavaDoc doclint on Java 8. It's annoying.
+    if (JavaVersion.current().isJava8Compatible()) {
+        allprojects {
+            tasks.withType(Javadoc) {
+                options.addStringOption('Xdoclint:none', '-quiet')
+            }
+        }
+    }
+
+    artifacts {
+        archives javadocJar, sourcesJar, shadowJar
+    }
+
+    publishing {
+        publications {
+            mavenJava(MavenPublication) {
+                artifactId archivesBaseName
+                from components.java
+                pom {
+                    name = 'minio'
+                    packaging = 'jar'
+                    description = 'MinIO Java SDK for Amazon S3 Compatible Cloud Storage'
+                    url = 'https://github.com/minio/minio-java'
+                    inceptionYear = '2015'
+
+                    scm {
+                        connection = 'scm:git:git@github.com:minio/minio-java.git'
+                        developerConnection = 'scm:git:git@github.com:minio/minio-java.git'
+                        url = 'http://github.com/minio/minio-java'
+                    }
+
+                    licenses {
+                        license {
+                            name = 'The Apache License, Version 2.0'
+                            url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                        }
+                    }
+
+                    developers {
+                        developer {
+                            id = 'minio'
+                            name = 'MinIO Inc.'
+                            email = 'dev@min.io'
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    signing {
+        if (project.properties.containsKey('signing.keyId')) {
+            sign publishing.publications.mavenJava
+        }
+    }
+
+    nexusStaging {
+        packageGroup = group
+        stagingProfileId = '9b746c9f8abc1'
+        username = project.properties['nexusUsername']
+        password = project.properties['nexusPassword']
+    }
+
+    nexusPublishing {
+        repositories {
+            sonatype {
+                username = project.properties['nexusUsername']
+                password = project.properties['nexusPassword']
+            }
+        }
+    }
+}
+
+project(':minio-admin') {
```

Comment:
I just copied and pasted what `:api` had

Reply:
I guess, `Gradle` might have a way to use common configurations. You could avoid repeated configurations.

Reply:
I gave this a shot but I'm a bit of a gradle n00b -- I tried defining a function, but the problem is it uses the global namespace (vs the task). Thus it breaks, as some things get re-used, when they're not expected to. Most notably, around the maven publish jar plugin.

Reply:
Got it